include/targets/hip/random_quda.h

```diff
@@ -0,0 +1,152 @@
+#pragma once
```

Comment:
There's a lot of repeated boilerplate in here and the `CUDA` equivalent.  While of course the `hiprandState` vs `cuRNGState` needs to be in the specialization, why not move the class `RNG` to the generic header?  We could make the class templated, with the type used to instantiate the class either `cuRNGState` or `hiprandState` in the respective target specialization?